src/Items/ItemSlab.h

```diff
@@ -63,11 +69,28 @@ class cItemSlabHandler:
 				((ClickedBlockMeta & 0x08) != 0)
 			)
 			{
+				a_World.BroadcastSoundEffect(PlaceSound, a_BlockX + 0.5, a_BlockY + 0.5, a_BlockZ + 0.5, Volume, Pitch);
 				return a_Player.PlaceBlock(a_BlockX, a_BlockY, a_BlockZ, m_DoubleSlabBlockType, ClickedBlockMeta & 0x07);
 			}
 		}
 
+		// Checking the type of block that should be placed
+		AddFaceDirection(a_BlockX, a_BlockY, a_BlockZ, a_BlockFace);
```

Comment:
I'd have preferred if you used a separate set of coords, `PlacedBlockX`, `PlacedBlockY` and `PlacedBlockZ`, and keep the values of `a_BlockX` et al.